docs/developer-guide/cstor-volume.md

```diff
@@ -0,0 +1,189 @@
+# cStor Volume Resource Organization
+
+## Introduction
+
+It is recommended to go through the cStor Pool resource organisation [doc](cstor-pool.md) before reading this document.
+
+CStor CSI volumes are created by creating a PVC in a Kubernetes cluster which references a CSPC based OpenEBS storage 
+class. A CSPC based storage class is simply a storage class that refers a CStorPoolCluster. A CStor CSI volume is 
+collection of 1 or more volume replica that gets ditributed across the CStorPoolInstances of a CStorPoolCluster. In this 
+sense, CStor volumes provide a replicated storage.
+
+To understand more on how CSI works please refer to the following document: 
+
+https://github.com/container-storage-interface/spec/blob/master/spec.md
+
+Following are the list of custom resources that enables the cStor CSI volume features:
+- CStorVolume(cv)
+- CStorVolumeReplica(cvr)
+- CStorVolumeConfig(cvc)
+- CStorVolumePolicy(cvp,policy)
+- CStorVolumeAttachment(cva)
+
+Following are the list of native Kubernetes resources that enables the cStor CSI volume features:
+- Service (Known as target service)
+- Deployment (Known as cStor Target)
+
+# cStor CSI Driver Overview
+
+OpenEBS cStor CSI driver(or plugin) follows a cetralized plugin architecture. In centralized plugin architeture, two 
+componenets are deployed in following manner:
+1. **Controller Plugin:** It is deployed as a deployment controller and scheduled to run on the master node of 
```

Comment:
Controller plugin can run anywhere not only in master.

Reply:
It is deployed as a sidecar container in statefulset with other CSI specific containers